src/test/java/VASSAL/tools/ipc/IPCMessageDispatcherTest.java

```diff
@@ -45,7 +45,7 @@ public void testRun() throws IOException, InterruptedException {
         exactly(msg.length+1).of(out).flush();
 
         oneOf(out).writeObject(fin);
-        exactly(2).of(out).close();
+        exactly(1).of(out).close();
```

Comment:
Reason for this is, the code originally did a `close()` inside a try-catch, and then a second `close()` in the finally block. And the test expected this behavior.

This is not a very clean way of closing streams, if the `close()` failed once, why would it work the second time around? And if we try it twice just in case, why twice, why not 4, 5 or 37 times, just to be on the safe side?

Reply:
while (!isClosedYet()) close();

Reply:
The reason for checking that close is called twice is that under normal operation, `close()` really was called twice---once by calling `close()` directly and once by `IOUtils.closeQuietly()`. If it's called some other number of times, the test should fail.

After your change, once is the correct number of times; but twice wasn't wrong before.